compatibility-tests/compile-fail/tests/ui/duplication-source-from-attributes.stderr

```diff
@@ -0,0 +1,11 @@
+error: Multiple `source` attributes are not supported on a field
+ --> $DIR/duplication-source-from-attributes.rs:9:18
+  |
+9 |          #[snafu(source(from(EnumError, Box::new)))]
+  |                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+error: Multiple `source(from)` attributes are not supported on a field
+ --> $DIR/duplication-source-from-attributes.rs:9:18
+  |
+9 |          #[snafu(source(from(EnumError, Box::new)))]
```

Comment:
It's because I used an `AtMostOne` called `source_attrs` for `source` tracking, and another called `transformations` for `source(from)` tracking.  When hitting a `source(from)` I need to add to both:

https://github.com/shepmaster/snafu/blob/88ee0375ac9ec84beb2fd25982e4d10ef057dd37/snafu-derive/src/lib.rs#L536-L537

This is so we can understand that a `source(from)` also counts as a source field, rather than allowing two source fields just because one uses `source` (or is named "source") and one uses `source(from)`.

An alternative would be to use a single list of source attributes, and track whether it was a transformation (`source(from)`) inside of each list element.  Currently `source_attrs` is just storing `()` as its data element, so maybe that could be an `Option` of the transformation data, for example.  I can try that.

Reply:
Oh! I missed that one was `source` and one was `source(from)`; that's fine, carry on!

Reply:
On balance, I think the "duplicate" is confusing and you were right to point it out.  It's caused by the same line of the user's code, and looks almost identical, and was only there because of an implementation detail.  I don't think it added any clarity to the presented errors.

Plus, the code is simpler without it, modulo some Rust 1.18 complaints.  I'm going to propose it in the next push and see what you think.  :)